src/main/java/mate/academy/spring/controller/ShoppingCartController.java

```diff
@@ -0,0 +1,49 @@
+package mate.academy.spring.controller;
+
+import mate.academy.spring.mapper.DtoResponseMapper;
+import mate.academy.spring.mapper.impl.response.ShoppingCartResponseMapper;
+import mate.academy.spring.model.MovieSession;
+import mate.academy.spring.model.ShoppingCart;
+import mate.academy.spring.model.User;
+import mate.academy.spring.model.dto.response.ShoppingCartResponseDto;
+import mate.academy.spring.service.MovieSessionService;
+import mate.academy.spring.service.ShoppingCartService;
+import mate.academy.spring.service.UserService;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PutMapping;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.RestController;
+
+@RestController
+@RequestMapping("/shopping-carts")
+public class ShoppingCartController {
+    private final MovieSessionService movieSessionService;
+    private final ShoppingCartService shoppingCartService;
+    private final UserService userService;
+    private final DtoResponseMapper<ShoppingCartResponseDto, ShoppingCart> responseMapper;
+
+    public ShoppingCartController(MovieSessionService movieSessionService,
+                                  ShoppingCartService shoppingCartService,
+                                  UserService userService,
+                                  ShoppingCartResponseMapper responseMapper) {
```

Comment:
you passing ShoppingCartResponseMapper to initialize your
 DtoResponseMapper<ShoppingCartResponseDto, ShoppingCart>
 i think better to also pass
  DtoResponseMapper<ShoppingCartResponseDto, ShoppingCart>